src/components/BrainfuckEngine.vue

```diff
@@ -249,33 +298,56 @@ export default {
       this.interpreter.resume();
       this.isPausing = false;
     },
-    input: function () {
-      for (let ch of this.inputText) {
+    addText: function (text) {
+      for (let ch of text) {
         this.interpreter.addInput(ch.charCodeAt(0));
       }
-      this.inputHistory += this.inputText + "\n";
+    },
+    addCharCode: function (charCode) {
+      this.interpreter.addInput(charCode);
+    },
+    input: function () {
+      this.addText(this.inputText);
+      this.inputHistory = [
+        ...this.inputHistory,
+        { type: "text", val: this.inputText },
+      ];
```

Comment:
```suggestion
      this.inputHistory.push({
        type: "text",
        val: this.inputText,
      });
```
In this case, the spread syntax inefficient and redundant.